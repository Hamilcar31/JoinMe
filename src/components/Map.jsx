import React from 'react';
import {View} from 'react-native';
import MapView from 'react-native-maps';
import window from '../constants/Layout.js';

export default () => {
    return (
        <View style={{flex: 1}} >
            <MapView style={window.window} />
        </View>
    );
}