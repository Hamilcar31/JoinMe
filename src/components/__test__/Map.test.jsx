/* eslint-disable react/prop-types */
import React from 'react';
import renderer from 'react-test-renderer';

import Map from '../Map';

jest.mock('react-native-maps', () => {
  // eslint-disable-next-line global-require
  const { View } = require('react-native');
  const MockMapView = (props) => {
    const { children } = props;
    return <View>{children}</View>;
  };
  const MockMarker = (props) => {
    const { children } = props;
    return <View>{children}</View>;
  };
  return {
    __esModule: true,
    default: MockMapView,
    Marker: MockMarker,
  };
});


describe('Map component', () => {
  it('with defaults parameters', () => {
    // given
    const map = <Map />;

    // when
    const tree = renderer.create(map).toJSON();

    // then
    expect(tree).toMatchSnapshot();
  });
});
