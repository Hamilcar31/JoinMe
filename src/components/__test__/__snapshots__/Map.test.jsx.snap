// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component with defaults parameters 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View />
</View>
`;
